Update email section of INSTALL.md about account_threepid_delegates (#6272)
matrix-org/synapse: INSTALL.md

## Email

It is desirable for Synapse to have the capability to send email. This allows
Synapse to send password reset emails, send verifications when an email address
is added to a user's account, and send email notifications to users when they
receive new messages. 

To configure an SMTP server for Synapse, modify the configuration section
headed `email`, and be sure to have at least the `smtp_host`, `smtp_port`
and `notif_from` fields filled out.  You may also need to set `smtp_user`,
`smtp_pass`, and `require_transport_security`.

> If email is not configured password reset, registration and notifications via

Reviewer: ```suggestion
If email is not configured, password reset, registration and notifications via
```

## Email

It is desirable for Synapse to have the capability to send email. This allows

> Synapse to send password reset emails, register users via their email address,

Reviewer: ```suggestion
Synapse to send password reset emails, send verifications when an email address is added to a user's account,
```

and send email notifications to users when they receive new messages. 

> Email can be set up in two ways, either by giving Synapse SMTP details

Reviewer: ```suggestion
Email can be set up either by giving Synapse SMTP details
```

> (recommended), or delegating another server to send emails on your behalf. Note

Reviewer: sending emails can't be delegated. *Verification* can be delegated, but that's only one use of emails.

that the latter will be removed in a future Synapse version.

To configure an SMTP server for Synapse, modify the configuration section
headed `email`, and be sure to have at least the `smtp_host`, `smtp_port`
and `notif_from` fields filled out.  You may also need to set `smtp_user`,
`smtp_pass`, and `require_transport_security`.

> To delegate an external server to handle email sending, modify the

Reviewer: frankly I'd get rid of this, given it's deprecated anyway.

Author: Yeah, probably more relevant to existing installs instead of new ones.
